src/libraries/System.Net.Security/src/System/Net/Security/SslSessionsCache.cs

```diff
@@ -125,9 +125,9 @@ public bool Equals(SslCredKey other)
             var key = new SslCredKey(thumbPrint, (int)sslProtocols, isServer, encryptionPolicy);
 
             SafeCredentialReference? cached;
-            if (!s_cachedCreds.TryGetValue(key, out cached) || cached.IsClosed || cached.Target.IsInvalid)
+            if (!s_cachedCreds.TryGetValue(key, out cached) || cached.Target.IsClosed || cached.Target.IsInvalid)
```

Comment:
When a SafeCredentialReference is disposed, it's null'ing out its target:
https://github.com/dotnet/runtime/pull/36875/files#diff-d9a8c0653d499676a48062a0a80b2018R46
Doesn't that mean this `cached.Target.IsClosed` could null ref?

Reply:
I was about to fix this as part of #2353

Reply:
But aren't you introducing the problem here?

Reply:
I think the issue was there before too as `cached.IsClosed` was always false fo the code reached `cached.Target.IsInvalid` condition check which would fail in the same way.

I added the changes from  #2353 so the null reference issue should be fixed.